qa/L0_additional_dependency_dirs/test.sh

```diff
@@ -0,0 +1,269 @@
+#!/bin/bash
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REPO_VERSION=${NVIDIA_TRITON_SERVER_VERSION}
+if [ "$#" -ge 1 ]; then
+    REPO_VERSION=$1
+fi
+if [ -z "$REPO_VERSION" ]; then
+    echo -e "Repository version must be specified"
+    echo -e "\n***\n*** Test Failed\n***"
+    exit 1
+fi
+
+if [[ -v WSL_DISTRO_NAME ]] || [[ -v MSYSTEM ]]; then
+    TRITON_DIR=${TRITON_DIR:=c:/tritonserver}
+    SERVER=${SERVER:=c:/tritonserver/bin/tritonserver.exe}
+    BACKEND_DIR=${BACKEND_DIR:=c:/tritonserver/backends}
+    MODELDIR=${MODELDIR:=c:/}
+    TRITONSERVER_IPADDR=${TRITONSERVER_IPADDR:="localhost"}
+else
+    echo -e "Test is not currently supported for Linux"
+    exit 1
+fi
+
+# FIXME: TPRD-244 - Do not use hard-coded dependency paths when TRT models
+# are being regularly generated on Windows.
+DEPENDENCY_PATH="C:/ci_test_deps/24.07"
+STALE_DEPENDENCY_PATH="C:/ci_test_deps/24.05"
+CUSTOM_DEPENDENCY_DIR=${MODELDIR}/custom_dependency_location
+STALE_DEPENDENCY_DIR=${MODELDIR}/stale_dependency_location
+TRT_MODEL_DIR="C:/tmp/24.07_trt_models"
```

Comment:
I think this feature is agnostic to the backend, so having TRT models in this test will probably make it more painful to maintain than necessary. In your follow-up ticket TPRD-244, I would suggest revisiting this. 

For example running CI with the `BACKENDS="python"` or something would have no effect on this test and fail looking for TRT models.

Instead I think TRT backend specific tests could add a test case for this feature if there's a specific TRT related flow it might be used for, and this test could be generic to something easier or more portable like a python model.

This would make it easier to compartmentalize running the tests on scenarios where it would be painful to get the TRT models for, or migrate to other runners, other versions, etc without compatibility issues.

Reply:
If the main use case driving the feature is intended for use with TRT backend for now, it's fine - just food for thought.

Reply:
I chose TRT for 2 reasons: 
1. The TRT backend was the motivating backend for this feature (as you say)
2. The backend requires implicit dependencies that are expected to exist outside of the backend directory.

The python backend does not depend on any implicit dependencies outside of what exists in its corresponding backend directory and those dependencies _must_ remain in that folder since their path relative to `libtriton_python.so` matters. I agree that having any other than TRT would be more maintainable, but I'd have to think on how I'd introduce an additional dependency to the python backend specifically for this test.